service/storage_service.cc

```diff
@@ -499,12 +499,10 @@ future<> storage_service::join_token_ring(cdc::generation_service& cdc_gen_servi
                 bootstrap_tokens = boot_strapper::get_bootstrap_tokens(tmptr, _db.local().get_config(), dht::check_token_endpoint::yes);
             }
         } else {
-            auto replace_addr = get_replace_address();
-            assert(replace_addr);
-            if (*replace_addr != get_broadcast_address()) {
```

Comment:
Maybe `assert(replace_address)`?

Reply:
Or change `if (is_replacing())` to `if (replace_address)` since it's set iff the former is true.

Reply:
Sounds good.